compiler/modules/bir/verify.bal

```diff
@@ -442,10 +442,26 @@ function verifyCall(VerifyContext vc, CallInsnBase insn) returns err:Internal? {
         return verifyFunctionCallArgs(vc, func.value.signature.paramTypes, insn);
     }
     t:SemType funcTy = func.semType;
-    t:FunctionAtomicType atomic = <t:FunctionAtomicType>t:functionAtomicType(vc.typeContext(), funcTy);
-    // TODO: in the non-atomic case verify restParamIsList is not set
-    t:FunctionSignature signature = t:functionSignature(vc.typeContext(), atomic);
-    return verifyFunctionCallArgs(vc, signature.paramTypes, insn);
+    t:FunctionAtomicType? atomic = t:functionAtomicType(vc.typeContext(), funcTy);
```

Comment:
I think we should split this function into direct and indirect cases.

Reply:
Fixed